builtin/logical/pki/fields.go

```diff
@@ -454,6 +454,13 @@ on revocation entries. This helps increase the performance of CRL building
 and OCSP responses.`,
 	}
 
+	fields["tidy_expired_issuers"] = &framework.FieldSchema{
+		Type: framework.TypeBool,
+		Description: `Set to true to automatically remove expired issuers
+past the issuer_safety_buffer. No keys will be removed as part of this
```

Comment:
Nope, I'm punting on that particular problem. Issuers make the mount slow (chain building, CRL building, ...), keys are always fetched by reference and don't make the mount slow.

Reply:
Yes, see the log message below:

https://github.com/hashicorp/vault/pull/17823/files#diff-4d3f13dea8cac8ec87e6f15d17a6bbbe798129d171059e492303070ce3903d9cR518

It contains the key ID from the issuer, allowing them to make a call from the log messages as to whether to remove that key or not (and the system will prevent that if its in use by other still-present issuers).

In the general case of tidying keys as well though, we don't necessarily presently store when/how a key was created, so we don't know if there's an outstanding, say, ICA issuance going on that we don't know about, or if it was from a deleted issuer.

Reply:
No, I don't want to automatically remove keys, but I don't want operators to get in a case where they'd like to remove an old key but are confused which issuer it once applied to.

Reply:
Would you recommend something else? Another migration to associate issuers on keys, and/or a tidy step to associate a since-deleted issuer's identifier(s) back to the key?

The problem is a number of issuers won't necessarily have meaningful names, so logging the full certificate is perhaps the best approach here -- unless we _do_ want to retain the issuer's certificate in the system and associate removed issuers' certificates with the key entry. Then they could read the key and see the list of since-deleted issuers which used that key... It wouldn't impact perf anymore but still wouldn't be ideal to interact with (as they'd likely have to parse the certificate to understand which one it was).

Reply:
No, I'd just be happy if the key_id was in the output of the issuer tidy, so they could grep it out and delete it by hand.